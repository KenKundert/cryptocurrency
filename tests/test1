#!/bin/csh -f

../cryptocurrency --config-dir test1_config BTC=10k ETH=100 BCH=1k zeus hermes persephone > test1.out

cmp test1.expected test1.out
if ($status) then
    vdiff test1.out test1.expected
else
    echo test1 passed
endif
