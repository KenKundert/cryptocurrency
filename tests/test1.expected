Override: USD = 10000.0*BTC
Override: USD = 100.0*ETH
Override: USD = 1000.0*BCH

Zeus's Holdings:
        6 BTC = $60k    89%    price=$10k per BTC      cost=$2.68k for 10 BTC      gain=273%
     26.2 ETH = $2.62k  4%     price=$100 per ETH
        5 BCH = $5k     7%     price=$1k per BCH
        Total = $67.6k         152% gain on $26.8k investment

Hermes's Holdings:
        1 BTC = $10k    99%    price=$10k per BTC
        1 ETH = $100    1%     price=$100 per ETH
        Total = $10.1k

Persephone's Holdings:
        1 BTC = $10k    99%    price=$10k per BTC
        1 ETH = $100    1%     price=$100 per ETH
        Total = $10.1k
